src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -129,10 +136,47 @@ class ZipFileSystem extends FileSystem {
     final boolean supportPosix;
     private final UserPrincipal defaultOwner;
     private final GroupPrincipal defaultGroup;
+    // Unmodifiable set.
     private final Set<PosixFilePermission> defaultPermissions;
 
     private final Set<String> supportedFileAttributeViews;
 
+    // If it's decided to try and make access mode common to other file systems,
+    // this could exist somewhere common, but if it's definitely never going to
+    // be shared, it could be made public here.
+    private enum AccessMode {
+        // Creates a file system for read-write access.
+        READ_WRITE("readWrite"),
+        // Creates a file system for read-only access.
+        READ_ONLY("readOnly");
+
+        private final String label;
+
+        AccessMode(String label) {
+            this.label = label;
+        }
+
+        // Parses the file system permission from an environmental parameter. While
+        // the FileSystemAccessMode is private, we don't need to check if it was
+        // given as an enum value. Returns null to indicate default behaviour.
+        static AccessMode from(Object value) {
+            switch (value) {
+                case null -> {
+                    return null;
+                }
+                case String label when READ_WRITE.label.equals(label) -> {
+                    return READ_WRITE;
+                }
+                case String label when READ_ONLY.label.equals(label) -> {
+                    return READ_ONLY;
+                }
+                default -> {
+                }
```

Comment:
Since we don't allow for any other value, I think moving the `throw new IllegalArgumentException` from outside of the switch into this `default` case might be suitable.